File: src/ClientApp/src/models/SearchRecipesRequest.ts
import type { RecipesSearchParams } from '@/api/data-contracts';
import Choices from '@/models/Choices';
import RecipeStoreHelpers from './RecipeStoreHelpers';

export default class SearchRecipesRequest implements RecipesSearchParams {
  public name = '';

  public categories = [];

  public isForMealPlanning = null;

  public sortBy = RecipeStoreHelpers.sortOptions[0].value;

  public isPagingEnabled = true;

  public page = 1;

  public take = Choices.defaultPaginationTake.value;
}

File: src/FoodStuffs.Model/Search/Recipes/Models/SearchRecipesRequest.cs
using VoidCore.Model.Responses.Collections;

namespace FoodStuffs.Model.Search.Recipes.Models;

public record SearchRecipesRequest(
    string? SearchText,
    int[]? CategoryIds,
    bool MatchAllCategories,
    int[]? GroceryItemIds,
    bool MatchAllGroceryItems,
    bool? IsForMealPlanning,
    string? SortBy,
    string? RandomSortSeed,
    bool IsPagingEnabled,
    int Page,
    int Take) : IPaginatedRequest;
